src/Effectful/HPQTypes.hs

```diff
@@ -48,6 +54,8 @@ data DB :: Effect where
 
 type instance DispatchOf DB = 'Dynamic
 
+-- This instance is not necessary to make the adapter work, but it can be useful
```

Comment:
But it is necessary, isn't it? The `DBEff` is an internal implementation detail.

Reply:
What I meant is that we can remove this instance and the rest of the module still works.  But you are right, the comment is pointless, the library would be unusable without this instance (already the tests don't compile without it).

Comment removed.